snuba/migrations/groups.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from importlib import import_module
+from typing import Sequence
+
+from snuba.migrations.migration import Migration
+
+
+class MigrationGroup(Enum):
+    SYSTEM = "system"
+    EVENTS = "events"
+
+
+class GroupLoader(ABC):
+    """
+    Provides the list of migrations associated with an group and a loader that returns
+    the requested migration.
+    """
+
+    @abstractmethod
+    def get_migrations(self) -> Sequence[str]:
+        """
+        Returns the list of migration IDs in the order they should be executed.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def load_migration(self, migration_id: str) -> Migration:
+        raise NotImplementedError
+
+
+class DirectoryLoader(GroupLoader):
+    """
+    Loads migrations that are defined as files of a directory. The file name
+    represents the migration ID.
+    """
+
+    def __init__(self, module: str) -> None:
+        self.__module = module
+
+    def get_migrations(self) -> Sequence[str]:
```

Comment:
It would be cool (later) to make this return a lazily loaded Iterable of migrations, like a generator, so you do not need to expose the migration names.

Reply:
I'll keep this in mind. I suspect we'll still need the migration names somewhere though. Migrations are referenced by name in the migrations table and it's probably the responsibility of the runner to reconcile this list with migration state from the db.